kotlinx-coroutines-test/common/src/TestScope.kt

```diff
@@ -170,6 +187,9 @@ internal class TestScopeImpl(context: CoroutineContext) :
     private val uncaughtExceptions = mutableListOf<Throwable>()
     private val lock = SynchronizedObject()
 
+    override val backgroundWorkScope: CoroutineScope =
+        CoroutineScope(coroutineContext + SupervisorJob() + BackgroundWork)
```

Comment:
This reveals that we need to improve the name/framing.

The purpose of this scope is not the fuzzy notion of background work, but, specifically, for coroutines that, from the perspective of the test, are always there. They are only cancelled so that resources are not leaked; semantically, they keep running for the whole duration of the test. I would even expect such background work to be launched outside of the test definition, in fields of the test class etc. So, if something significant happens on cancellation of a background task, using this scope is not the proper way to check that.

Reply:
Even with a better name, the problem will still exist. It would be nice to report _any_ exceptions that happen within the scope

Reply:
Not sure about that.

Given the more clear definition of background work, we see that, typically, the background work executes in places that suffer from the same problem, only in production. For example, `viewModelScope` uses `SupervisorJob`, and something crashing in `GlobalScope` is notoriously elusive.

So, let's consider the possibilities.
* The background code never crashes silently in production. Ok, fair enough, nothing to discuss.
* It may crash silently in production, but in a controllable manner that the foreground code expects. Then we would just complicate the tests with an unneeded error-handling ceremony if we forced the error to surface.
* It may crash silently in production, and this is unexpected and should be tested for. Well, why is that not a separate test then?

Reply:
After an internal discussion, we arrived at this solution: do keep `SupervisorJob`, because it allows the background tasks to fail independently of the test code and one another, which would, in our opinion, lead to the least surprising behavior, but *do* report all the exceptions that it is possible for us to observe. For now, we will assume that the actual production code does not rely on `async` and `produce` not surfacing the errors until queried. If this helps surface some errors, great. If it turns out that people *are* using the fact that exceptions are not reported, we will learn about this with actual use cases. In general, we follow the rule of thumb that exceptions are probably something that's not to be unhandled.